service/src/test/java/tds/session/services/impl/SessionServiceImplTest.java

```diff
@@ -77,4 +89,145 @@ public void shouldReturnSessionAssessment() {
+            .build();
+
+        // What the session will look like after mockSessionRepository.pause() is called
+        Session mockUpdatedSession = new Session.Builder()
+            .withId(sessionId)
+            .withStatus("closed")
+            .withDateChanged(Instant.now().plus(750000))
+            .withDateEnd(Instant.now().plus(750000))
+            .withProctorId(proctorId)
+            .withBrowserKey(browserKey)
+            .build();
+
+        PauseSessionRequest request = new PauseSessionRequest(proctorId, browserKey);
+
+        when(mockSessionRepository.findSessionById(mockOpenSession.getId()))
+            .thenReturn(Optional.of(mockOpenSession))
+            .thenReturn(Optional.of(mockUpdatedSession));
+        doNothing().when(mockSessionRepository).pause(mockOpenSession.getId(), "closed");
+        doNothing().when(mockExamService).pauseAllExamsInSession(mockOpenSession.getId());
+
+        Response<PauseSessionResponse> response = service.pause(mockOpenSession.getId(), request);
+
+        verify(mockExamService).pauseAllExamsInSession(mockOpenSession.getId());
+        verify(mockSessionRepository).pause(mockOpenSession.getId(), "closed");
+        verify(mockSessionRepository, times(2)).findSessionById(mockOpenSession.getId());
+
+        assertThat(response.getData()).isNotNull();
+        assertThat(response.getErrors()).hasSize(0);
+        PauseSessionResponse pauseSessionResponse = response.getData().get();
+        assertThat(pauseSessionResponse.getStatus()).isEqualTo(mockUpdatedSession.getStatus());
+        assertThat(pauseSessionResponse.getSessionId()).isEqualTo(mockUpdatedSession.getId());
+        assertThat(pauseSessionResponse.getDateChanged()).isNotNull();
+        assertThat(pauseSessionResponse.getDateEnded()).isNotNull();
+    }
+
+    @Test
+    public void shouldRespondWithSessionClosedValidationErrorWhenPausingAClosedSession() {
+        long proctorId = 1L;
+        UUID sessionId = UUID.randomUUID();
+        UUID browserKey = UUID.randomUUID();
+
+        Session mockClosedSession = new Session.Builder()
+            .withId(sessionId)
+            .withStatus("closed")
+            .withDateChanged(Instant.now().minus(600000))
+            .withDateEnd(Instant.now().minus(600000))
+            .build();
+
+        PauseSessionRequest request = new PauseSessionRequest(proctorId, browserKey);
+
+        when(mockSessionRepository.findSessionById(mockClosedSession.getId())).thenReturn(Optional.of(mockClosedSession));
+        doNothing().when(mockExamService).pauseAllExamsInSession(mockClosedSession.getId());
+
+        Response<PauseSessionResponse> response = service.pause(sessionId, request);
+
+        verify(mockExamService, times(0)).pauseAllExamsInSession(mockClosedSession.getId());
+        verify(mockSessionRepository, times(0)).pause(mockClosedSession.getId(), "closed");
+        verify(mockSessionRepository, times(1)).findSessionById(mockClosedSession.getId());
```

Comment:
I think lines 161 and 162 could be changed to:

```
verify(mockSessionRepository).findSessionById(mockClosedSession.getId());
verifyNoMoreInteractions(mockSessionRepository);
```

Basically there should be that one interaction and no others since the session and request aren't in a valid state.